app/src/androidTest/java/com/github/h3lp3rs/h3lp/presentation/PresIrrelevantActivityTest.kt

```diff
@@ -3,50 +3,201 @@ package com.github.h3lp3rs.h3lp.presentation
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions.*
 import androidx.test.espresso.assertion.ViewAssertions.matches
 import androidx.test.espresso.intent.Intents.*
 import androidx.test.espresso.intent.matcher.IntentMatchers.*
 import androidx.test.espresso.matcher.ViewMatchers.*
-import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
-import com.github.h3lp3rs.h3lp.MainPageActivity
-import com.github.h3lp3rs.h3lp.R
-import org.hamcrest.core.AllOf.*
-import org.junit.Rule
+import com.github.h3lp3rs.h3lp.*
+import com.github.h3lp3rs.h3lp.preferences.Preferences
+import com.github.h3lp3rs.h3lp.preferences.Preferences.Companion.Files.*
+import com.github.h3lp3rs.h3lp.preferences.Preferences.Companion.USER_AGREE
+import com.github.h3lp3rs.h3lp.preferences.Preferences.Companion.clearAllPreferences
+import org.hamcrest.Matchers.*
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
+import androidx.test.espresso.action.Press
+
+import androidx.test.espresso.action.CoordinatesProvider
+
+import androidx.test.espresso.action.Tap
+
+import androidx.test.espresso.action.GeneralClickAction
+
+import androidx.test.espresso.ViewAction
+
+
+
 
 @RunWith(AndroidJUnit4::class)
 class PresIrrelevantActivityTest {
-    @get:Rule
-    val testRule = ActivityScenarioRule(PresIrrelevantActivity::class.java)
 
-    @Test
-    fun successfulDisplay() {
-        onView(withId(R.id.pres3_textView1)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_textView2)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_textView3)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_textView4)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_textView5)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_textView6)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_imageView1)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_imageView2)).check(matches(isDisplayed()))
-        onView(withId(R.id.pres3_imageView3)).check(matches(isDisplayed()))
+    private fun alreadyAccepted() {
+        Preferences(PRESENTATION, ApplicationProvider.getApplicationContext()).setBool(USER_AGREE, true)
     }
 
-    @Test
-    fun successfulApprovalButton() {
+    // https://stackoverflow.com/questions/42390788/espresso-click-on-specific-words-of-text
+    // Since this is not a textView, espresso cannot click on text in it. So we manually find the
+    // right coordinates of the highlighted text.
+    private fun clickPercent(pctX: Float, pctY: Float): ViewAction {
+        return GeneralClickAction(
+            Tap.SINGLE,
+            { view ->
+                val screenPos = IntArray(2)
+                view.getLocationOnScreen(screenPos)
+                val w = view.width
+                val h = view.height

```

Comment:
Don't Know if here h and v are really useful

Reply:
Indeed, CC'd the SOF code.
Will refactor at next sprint.